itou/approvals/management/commands/bulk_update_from_file.py

```diff
@@ -32,74 +32,83 @@ def add_arguments(self, parser):
             help="Absolute path of the XLSX file to import",
         )
 
-        parser.add_argument(
-            "--company-id",
-            action="store",
-            dest="company_id",
-        )
-
-    def handle(self, wet_run, file_path, company_id, **options):
+    def handle(self, wet_run, file_path, **options):
         df = pd.read_excel(file_path)
         date_format = "%d/%m/%Y"
         df["num_pass_iae"] = df["PASS IAE"].str.replace(" ", "")
         df["debut_de_cddi"] = pd.to_datetime(df["Début de CDDI"], format=date_format).apply(lambda x: x.date())
+        df["siret"] = df["SIRET"].astype(str).str.replace(" ", "")
         df["pass_maj"] = False
-        df["commentaire"] = None
+        df["candidature_maj"] = False
+        df["commentaire PASS IAE"] = None
+        df["commentaire candidature"] = None
         df["date_debut_pass_iae"] = None
 
         job_applications_to_update = []
         approvals_to_update = []
         approvals_nb_with_errors = []
-        company = Company.objects.get(pk=company_id)
+        companies_qs = Company.objects.filter(siret__in=set(df["siret"].tolist())).values("pk", "siret")
+        companies_siret_to_pk = {company["siret"]: company["pk"] for company in companies_qs}
 
         approvals_qs = Approval.objects.filter(number__in=df["num_pass_iae"])
 
         for i, row in df.iterrows():
+            update_job_application = False
+            if not companies_siret_to_pk.get(row["siret"]):
+                df.loc[i, "commentaire PASS IAE"] = "SIRET vide. Pas de modification possible."
+                continue
+
             if row["num_pass_iae"] not in approvals_qs.values_list("number", flat=True):
```

Comment:
tu devrais créer un `approvals = {approval.pk: approval for approval in approvals_qs}`
Cela te permettrai d'éviter de faire des `values_list()` / `get()` à chaque ligne du fichier.